src/main/java/teammates/common/util/TimeHelper.java

```diff
@@ -325,68 +342,88 @@ private static String formatInstant(Instant instant, ZoneId timeZone, String pat
         return zonedDateTime.format(formatter);
     }
 
-    public static String formatDateTimeForSessions(Instant instant, ZoneId sessionTimeZone) {
-        return formatInstant(instant, sessionTimeZone, "EEE, dd MMM yyyy, hh:mm a z");
-    }
-
-    public static String formatDateTimeForDisambiguation(Instant instant, ZoneId zone) {
-        return formatInstant(instant, zone, "EEE, dd MMM yyyy, hh:mm a z ('UTC'Z)");
-    }
-
     /**
-     * Formats a date in the format d MMM h:mm a. Example: 5 May 11:59 PM
+     * Formats a datetime stamp from an {@code instant} including time zone name.
+     * Example: Sun, 01 Apr 2018, 11:21 PM SGT
+     *
+     * <p>Note: a datetime with time "12:00 PM" is specially formatted to "12:00 NOON"
+     * Example: Sun, 01 Apr 2018, 12:00 NOON SGT</p>
+     *
+     * @param instant         the instant to be formatted
+     * @param sessionTimeZone the time zone to compute local datetime
+     * @return the formatted datetime stamp string
      */
-    public static String formatDateTimeForInstructorHomePage(LocalDateTime localDateTime) {
-        return formatLocalDateTime(localDateTime, "d MMM h:mm a");
+    public static String formatDateTimeForDisplayFull(Instant instant, ZoneId sessionTimeZone) {
+        return formatInstant(instant, sessionTimeZone, "EEE, dd MMM yyyy, hh:mm a z");
     }
 
     /**
-     * Formats a date in the format d MMM yyyy. Example: 5 May 2017
+     * Formats a datetime stamp from an {@code instant} for DST disambiguation including time zone name and offset.
+     * Example: Sun, 01 Apr 2018, 11:23 PM SGT (UTC+0800)
+     *
+     * <p>This is used to generate the warning to instructors for datetimes that fall in DST overlaps.</p>
+     *
+     * <p>Note: a datetime with time "12:00 PM" is specially formatted to "12:00 NOON"
+     * Example: Sun, 01 Apr 2018, 12:00 NOON SGT (UTC+0800)</p>
+     *
+     * @param instant the interpreted instant to be formatted
+     * @param zone    the time zone causing the DST overlap
```

Comment:
Same here. The method might be useful for more than just DST overlaps even if that's currently the only use case.

Reply:
With these two review comments, should I also change the method name then? The current name is `formatDateTimeForDisambiguation` (similar to the original). This method is like `formatDateTimeForDisplayFull`, but adds the UTC offset to the end.

I renamed them so that the use of `full` makes more sense. There are now three similarly-named methods:

```bash
# no time zone data in formatted string
formatDateTimeForDisplay(LocalDateTime)

# originally formatDateTimeForDisplayFull(Instant, ZoneId)
# time zone short name in string, e.g. SGT
formatDateTimeForDisplay(Instant, ZoneId)

# originally formatDateTimeForDisambiguation(Instant, ZoneId)
# time zone short name and UTC offset in string, e.g. SGT (UTC+0800)
formatDateTimeForDisplayFull(Instant, ZoneId)
```

Reply:
Good point. The new names are good 👍